### ISSUE TITLE
Vocabulary is not exposed in the properties metadata

### ISSUE THREAD

**Issue by sabbyanandan** [spring-cloud-stream-app-starters/tensorflow#5]
Though there is the option to override and supply the [vocabulary](https://github.com/spring-cloud-stream-app-starters/tensorflow/blob/121327ef60cf7b020c8b8b160e77d4517d34bb95/spring-cloud-starter-stream-processor-twitter-sentiment/src/main/java/org/springframework/cloud/stream/app/twitter/sentiment/processor/TwitterSentimentProcessorProperties.java#L40), it doesn't show up in the properties metadata. And because it is not showing up, the `app info` and the dashboard in SCDF doesn't show that as a supported property.

**stream:**
```
sa-tweets=twitterstream | filter --expression=#jsonPath(payload,'$.lang')=='en' | twitter-sentiment --tensorflow.twitter.vocabulary=http://dl.bintray.com/big-data/generic/vocab.csv --model=http://dl.bintray.com/big-data/generic/minimal_graph.proto --tensorflow.model-fetch=output/Softmax | field-value-counter --field-name=sentiment --name=sentiment
```

![image](https://user-images.githubusercontent.com/1562654/31363255-06b210de-ad12-11e7-828e-4947e3e3f809.png)

**Comment by artembilan** [comment]
@sabbyanandan,

See my comment in the recently closed issue: https://github.com/spring-cloud-stream-app-starters/tensorflow/issues/4#issuecomment-331966859.

Maybe the bug appears in the released version but the current master is OK? More over I see the mentioned configuration properties class is white-listed: https://github.com/spring-cloud-stream-app-starters/tensorflow/blob/master/spring-cloud-starter-stream-processor-twitter-sentiment/src/main/resources/META-INF/spring-configuration-metadata-whitelist.properties.

Will be glad to screen share tomorrow to nail the bug :wink:

**Comment by sabbyanandan** [comment]
You're correct, @artembilan. I see the property in `1.0.0.BUILD-SNAPSHOT`. 

![image](https://user-images.githubusercontent.com/1562654/31405046-a8381e62-adb2-11e7-908f-660003e152b0.png)

@sobychacko: Any ideas why it is not in 1.0.0.M1/Celsius.M1?

**Comment by sobychacko** [comment]
@sabbyanandan I will take a look. Not sure why it doesn't show up.

**Comment by tzolov** [comment]
The problem seems related to the way `spring-cloud-app-starters-maven-plugins` search for the `‘META-INF/spring-configuration-metadata-whitelist.properties`.   The app-starters-maven-plugins search the classpath and picks the first whitelist properties file found. 
In the case of the tensorflow app there are 2 processors `tensorflow-processor` and `twitter-sentiment` where the later depends on the former!  So whitelist props for both processors are on the classpath. The maven plugin always picks the first one resulting on twitter-sentiment having the metadata for tensorflow-processor.

**Comment by tzolov** [comment]
I'm working on factoring out the common code in a common `spring-cloud-starter-stream-common-tensorflow` project used by both processors. This should solve the problem.
